command/agentproxyshared/auth/auth.go

```diff
@@ -478,10 +479,9 @@ func (ah *AuthHandler) Run(ctx context.Context, am AuthMethod) error {
 		}
 
 		metrics.IncrCounter([]string{ah.metricsSignifier, "auth", "success"}, 1)
-		// We don't want to trigger the renewal process for tokens with
-		// unlimited TTL, such as the root token.
-		if leaseDuration == 0 && isTokenFileMethod {
```

Comment:
Previously, this was a bug, and could occur for non-renewable tokens that happened to check it as their lease ticked down, causing re-authentication to not happen.

Reply:
Do we want to keep a check doesn't trigger lifetime watcher if the secret.Renewable == false?

Reply:
If the secret's not renewable, then the LifetimeWatcher still needs to get triggered given the current architecture, so that auth gets re-triggered at the correct time. If we need to authenticate again for any reason, we should run the lifetimewatcher -- the root token is the only token where we don't need to reauthenticate (since it never expires)

e.g. for a 10s TTL non-renewable token, it might look like this
- LifetimeWatcher triggered
- It waits for ~8.5 seconds, does not try to renew
- Triggers the done channel with no error
- Auth gets re-triggered

We're essentially making renewable tokens without the permission to renew follow the same path as the above ^